rclpy/rclpy/node.py

```diff
@@ -14,24 +14,30 @@
 
 import rclpy
 
+from rclpy.client import Client
 from rclpy.publisher import Publisher
-from rclpy.subscription import Subscription
 from rclpy.qos import qos_profile_default
+from rclpy.service import Service
+from rclpy.subscription import Subscription
 
 
 class Node:
 
     def __init__(self, handle):
+        self.clients = []
         self.handle = handle
+        self.publishers = []
+        self.services = []
```

Comment:
The main justification is that there is no executor/callback groups in python so storing them here allows `spin_once` to iterate over all these and populate/check the wait_set accordingly.

Another reason is that the `rcl_(client/publisher/service)_fini` functions need the node pointer to delete them. If we store a node pointer in all of these instances(client/publisher/service) and explicitely delete them in the destructor of each class the deletion order is not deterministic and result in segfaults. Storing them in the node allows to make sure that the node outlives every entity attached to it.